git.go

```diff
@@ -126,6 +127,15 @@ func (g *GitClient) Fetch(uri string, prNumber int, depth int) error {
 	return nil
 }
 
+// CheckOut
+func (g *GitClient) Checkout(branch string) error {
+	if err := g.command("git", "checkout", branch).Run(); err != nil {
```

Comment:
I don't think this will work, as the PR branch does not exist locally. So we need to this to do `git checkout -b <branch> <headsha>` so we check out the correct commit.

Reply:
As long as the remote `origin` exists, `git checkout` will create a new branch that reference the remote with the same branch name so it should work.

```
❯ git checkout install-something
Branch 'install-something' set up to track remote branch 'install-something' from 'origin'.
Switched to a new branch 'install-something'
```

But I would agree to have integration tests to check if this actually works or not, so I'm going to give it a try.

Reply:
How do I run the `e2e` tests btw?
I don't see any instructions of how to do it.

Reply:
The `e2e` tests are run with `task e2e`, and only requires that `GITHUB_ACCESS_TOKEN ` is set in the environment. The `put` tests will fail since your token won't have permissions to set status on my repository unfortunately, so I think you can skip `repo:status` permissions and just go for `public_repo` to test `check` and `get` 🤔

Reply:
Not an ideal situation btw, open to suggestions if there are better ways to solve this...

Reply:
This checkout will do check out for the head of the branch of the PR, not for the actual commit that triggered that pr?

Reply:
Yep @Nemeq! Need to fix that so we test the commit that we intend to set the status for 😅

Reply:
My bad for just naming the argument `branch`. Technically `git checkout` can checkout shas too so would that work better?
I can change `in.go` to checkout the HEAD sha, which should be the actual commit and maybe rename the argument to `sha` instead of `branch`.

The end result of all this will always end up with a stripped down git repo to be passed to tasks anyway so using the branch name of not doesn't seem to matter that much.

Reply:
I think when checking out just a commit, `git` will post a big info-blob about being in a detached state. Since we already have the name of the branch I think we can just include the branch and just do a `git checkout -b <branch> <sha>`?

Reply:
cool, i'll update the code to do that then.